workbox-config.js

```diff
@@ -0,0 +1,29 @@
+module.exports = {
+  globDirectory: 'web/',
+  globPatterns: [
+    '**/*.{html,md,json,ico,png,svg,eot,ttf,jpg,css,js,gif,woff,woff2,snap,webmanifest,rss-20130412,json-20170530,cgi,cgi-error,backup}',
+  ],
+  swDest: 'web/sw.js',
+  ignoreURLParametersMatching: [
+    /^utm_/,
+    /^fbclid$/,
+  ],
+  // Define runtime caching rules.
+  runtimeCaching: [{
+    // Match any request that ends with .png, .jpg, .jpeg or .svg.
+    urlPattern: /\.(?:png|jpg|jpeg|svg)$/,
```

Comment:
I'd like to understand the caching aspect better, I'm guessing this only applies to assets that live within the deployed app dirs somewhere and not to any requests for assets we make to external APIs (e.g. GIBS) during runtime?

Reply:
That's correct.  If you want to check out what cache is built, you can open Dev Tools > Application > Cache Storage (left menu) > Workbox-precache-v2.   There you will see all of the entries that are cached by the service worker.  As the app is being used and other layers are added and viewed, the are not added to the cache.

However, I will say that the cache that is created is fairly large (102 MB).  We could exclude directories if needed.

Here is some information about precaching with Workbox:  https://developers.google.com/web/tools/workbox/modules/workbox-precaching

Here is where I pulled some code for the `workbox-config.js`  https://developers.google.com/web/tools/workbox/guides/generate-service-worker/cli#adding_runtime_caching